circlemenu/src/main/java/com/szugyi/circlemenu/view/WheelLayout.java

```diff
@@ -0,0 +1,372 @@
+            if (children == null || item == null)
+                continue;
+            setChildrenValue(children, item);
+            itemIndexStartReversed++; // update Item index
+        }
+    }
+
+    @Override
+    public View getChildAt(int index) {
+        return super.getChildAt(index%getChildCount());
+    }
+
+    private void setChildrenValue(View children, WheelItem item) {
+        if (children instanceof TextView) {
+            ((TextView) children).setText(item.getName());
+            children.setTag(R.id.wheel_item_id,item);
+        }
+    }
+
+    public void addItems(List<? extends WheelItem> cats) {
+        if (cats == null || cats.isEmpty())
+            return;
+        if (mItems == null)
+            mItems = new ArrayList<>();
+        mItems.addAll(cats);
+        setPageCount();
+    }
+
+    private void setPageCount() {
+        mItemCount = mItems.size();
+    }
+
+    public int getItemCount(){
+        return mItems !=null ? mItems.size() : -1;
+    }
+
+    private WheelItem getItem(int index) {
+        if (mItems ==null || mItems.isEmpty() ||  index < 0)
+            return null;
+        index = index%mItemCount; // repeat displaying items on wheel
+        return mItems.get(index);
+    }
+
+    private void loadMore(boolean isNext){
+        if (onLoadMoreListener !=null)
+            onLoadMoreListener.loadMore(isNext);
+    }
+
+    public WheelItem getSelectedWheelItem(){
+        return getItemByView(getSelectedItem());
+    }
+
+    private WheelItem getItemByView(View child) {
+        return child !=null
+                && child.getTag(R.id.wheel_item_id) instanceof WheelItem
+                ? (WheelItem) child.getTag(R.id.wheel_item_id)
+                : null;
+    }
+
+
+    public void select(WheelItem itemName) {
+        if (itemName == null)
+            return;
+        select(itemName.getName());
+    }
+
+    public void select(String itemName) {
+        if (itemName == null)
+            return;
+        final View viewToCenter = getViewByItem(itemName);
+        if (viewToCenter !=null)
+            rotateViewToCenter(viewToCenter);
+    }
+
+    private View getViewByItem(String itemName) {
+        for (int i = 0; i < getChildCount(); i++){
+            final View child = getChildAt(i);
+            final WheelItem item = getItemByView(child);
+            if (item !=null && item.getName().equals(itemName))
+                return child;
+        }
+        return null;
+    }
+
+    @Override
+    public void onRotationFinished(View view) {
+        if (onRotateListener !=null)
+            onRotateListener.onRotationFinished(view);
+    }
+
+    public interface WheelItem{
+        String getName();
+        long getId();
+    }
+
+    public interface OnRotateListener extends OnRotationFinishedListener{
+        void onSettleRotation(float endDegree, long duration);
+        void onRotate(float degree);
+        void onStopAnimation();
```

Comment:
why not onRotationFinished?

Reply:
onRotationFinished called after animation finished, 
onSettleRotation > may be onAmimationStarted is more descriptive,